### PR TITLE
✨ Start metadata

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR introduces a subsystem for maintaining a reflection of git metadata into directories of text files and symlinks to make it easy to answer questions that the `git` CLI does not have an easy way to answer.  For example: "what branches --- if any --- contain commit 1a6140f3?"

## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
The payoff is shown at the end of the README: this makes it easy to answer questions like "is commit X in release Y?".

File: metadata/README.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+(venv) bash-3.2$ grep ^1a6140f3 branches/*
+branches/main:1a6140f3bc1422ea160479d2043555bd6fd4391d
+branches/start-metadata:1a6140f3bc1422ea160479d2043555bd6fd4391d
+
+(venv) bash-3.2$ grep ^1a6140f3 remotebranches/*
+remotebranches/origin-HEAD:1a6140f3bc1422ea160479d2043555bd6fd4391d
+remotebranches/origin-main:1a6140f3bc1422ea160479d2043555bd6fd4391d
+remotebranches/upstream-HEAD:1a6140f3bc1422ea160479d2043555bd6fd4391d
+remotebranches/upstream-main:1a6140f3bc1422ea160479d2043555bd6fd4391d
```
Comment (ezrasilvera) [inline metadata/README.md:199]: Do you collect data from all remotes ?

Comment (MikeSpreitzer) [reply metadata/README.md:199]: Yes.

**Comment by MikeSpreitzer** [comment]
This is not needed.  The following git commands cover the interesting capabilities.

- `git branch --contains $id`
- `git branch -r --contains $id`
- `git tag --contains $id`

**Pull request closed without merging**